test/python/synthesis/test_multi_controlled_synthesis.py

```diff
@@ -0,0 +1,384 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2025.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test synthesis algorithms for multi-controlled gates."""
+
+import unittest
+from test import combine
+import numpy as np
+from ddt import ddt, data
+
+from qiskit.quantum_info import Operator
+from qiskit.circuit import QuantumCircuit, Gate
+from qiskit.circuit.library import (
+    XGate,
+    RXGate,
+    RYGate,
+    RZGate,
+    PhaseGate,
+    YGate,
+    ZGate,
+    HGate,
+    SGate,
+    SdgGate,
+    TGate,
+    TdgGate,
+    SXGate,
+    SXdgGate,
+    UGate,
+)
+from qiskit.synthesis.multi_controlled import (
+    synth_mcx_n_dirty_i15,
+    synth_mcx_n_clean_m15,
+    synth_mcx_1_clean_b95,
+    synth_mcx_gray_code,
+    synth_mcx_noaux_v24,
+    synth_c3x,
+    synth_c4x,
+)
+from qiskit.circuit._utils import _compute_control_matrix
+from qiskit.quantum_info.operators.operator_utils import _equal_with_ancillas, matrix_equal
+from qiskit.transpiler import generate_preset_pass_manager
+
+from test import QiskitTestCase  # pylint: disable=wrong-import-order
+
+
+@ddt
+class TestMCSynthesisCorrectness(QiskitTestCase):
+    """Test correctness of synthesis methods for multi-controlled gates."""
+
+    @staticmethod
+    def mc_matrix(base_gate: Gate, num_ctrl_qubits: int):
+        """Return matrix for the MC gate with the given base gate and the number of control qubits."""
+        base_mat = base_gate.to_matrix()
+        return _compute_control_matrix(base_mat, num_ctrl_qubits)
+
+    def check_mc_synthesis(
```

Comment:
The test methods are typically called something like `assertSynthesisCorrect` or something, in case you'd like to be consistent. We use the same naming scheme in some other tests too

Reply:
Good point, will do!